Newsletters: add 348 (2025-04-04)

harding opened the pull request:
- [x] Write news item and remaining _changing consensus_ items @harding 
- [x] Write lede, update releases/RCs, add topic entries @harding 
- [x] Notable merges @Gustavojfe 

Sorry this PR is being opened in draft.  At the time of opening, with lots of content still missing, it's our third-longest regular newsletter ever.  By the time it's done, it'll almost certainly be longer than any other regular newsletter and also longer than several of our year-end newsletters.

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      the date at which someone with a quantum computer can begin
+      stealing bitcoins, but a specific date at which quantum-vulnerable
+      coins will be destroyed can be announced far in advance with
+      perfect precision.  That clear deadline will provide more
+      incentive for users to upgrade in time, ensuring fewer total coins
```
```suggestion
      perfect precision.  That clear deadline will
      incentivize users to move their coins in time, ensuring fewer total coins
```
(to me, "upgrade" sounds like installing new software, but I think what's meant is moving coins to QR-safe UTXOs)

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      they might do even if the rest of Bitcoiners prefer a different
+      outcome.
+
+    Lopp also provides several arguments against destruction of
```
```suggestion
    Lopp also provides several arguments against the destruction of
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      outcome.
+
+    Lopp also provides several arguments against destruction of
+    vulnerable Bitcoins, but he concludes in favor of destruction.
```
```suggestion
    vulnerable bitcoins, but he concludes in favor of destruction.
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+  several conversations examined how Bitcoiners could respond to quantum
+  computers becoming powerful enough to allow stealing bitcoins.
+
+  - *Should vulnerable bitcoins be destroyed?* Jameson Lopp [posted][lopp
+    destroy] to the Bitcoin-Dev mailing list several arguments for the
+    destruction of bitcoins vulnerable to quantum theft after an upgrade
```
What follows is an excellent summary of Lopp's post; I'm wondering if we should mention here that his proposal would be a soft fork (he mentions that several times)? Also, it might help future searches of the Optech archive to work in at least some of the words "burn", "unspendable", "freeze", or "frozen" somewhere here.

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    Lopp also provides several arguments against destruction of
+    vulnerable Bitcoins, but he concludes in favor of destruction.
+
+    Nagaev Boris [asks][boris timelock] about whether UTXOs that are
```
```suggestion
    Nagaev Boris [asks][boris timelock] whether UTXOs that are
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    computers existed).  If the UTXO contained a SHA256 commitment (or
+    other quantum-resistant commitment) to a preimage that had never
+    previously been revealed, then a multistep protocol for revealing
+    that preimage could be used to prevent quantum theft.  This is
```
```suggestion
    that preimage could prevent quantum theft.  This is
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+       spends).
+
+    3. When Alice is ready to spend, she privately creates her spending
+       transaction in the normal way but does not broadcast it.  She
```
```suggestion
       transaction normally but does not broadcast it.  She
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+       revealing it).  She waits for this transaction to be deeply
+       confirmed.
+
+    5. After she's absolutely sure her previous transaction can't
```
```suggestion
    5. After she's sure her previous transaction can't
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    - *Argument from common preference:* he believes most people would
+      prefer that their funds be destroyed rather than stolen by someone
+      with a fast quantum computer.  Especially, he argues, since the
+      thief will be it the set of "the few privileged folks who gain
```
```suggestion
      thief will be in the set of "the few privileged folks who gain
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    post][ruffing gfsig].  Although not discussed in the thread, we
+    believe the above protocol could be deployed as a soft fork.
+
+    Habovštiak argues that bitcoins which can be securely spent using
```
```suggestion
    Habovštiak argues that bitcoins that can be securely spent using
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+
+    Habovštiak argues that bitcoins which can be securely spent using
+    this protocol (e.g., their preimage hasn't already been revealed)
+    should not be destroyed even if the community decides in wants to
```
```suggestion
    should not be destroyed even if the community decides it wants to
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    quantum theft if keypath spending is later disabled.
+
+    In a separate thread (see next item), Pieter Wuille [notes][wuille
+    nonpublic] that UTXOs vulnerable to quantum theft also includes keys
```
```suggestion
    nonpublic] that UTXOs vulnerable to quantum theft also include keys
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+
+    In a separate thread (see next item), Pieter Wuille [notes][wuille
+    nonpublic] that UTXOs vulnerable to quantum theft also includes keys
+    which have not been used publicly but which are known to multiple
```
```suggestion
    that have not been used publicly but which are known to multiple
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+  - *Draft BIP for destroying quantum-insecure bitcoins:* Agustin Cruz
+    [posted][cruz qramp] to the Bitcoin-Dev mailing a [draft BIP][cruz
+    bip] that describes several options for a general process of
+    destroying bitcoins that are vulnerable to quantum theft (in the
+    event that becomes an expected risk).  Cruz argues, "enforcing a
```
```suggestion
    destroying bitcoins that are vulnerable to quantum theft (if
    that becomes an expected risk).  Cruz argues, "enforcing a
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+       also obtains some bitcoin secured by a quantum-secure signature
+       algorithm so that she can pay transaction fees.
+
+    4. In a transaction spending some of the quantum-secure bitcoins,
+       she creates a transaction that uses her private knowledge of a
```
In a transaction … creates a transaction?

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    Bitcoin."
+
+    Very little of the discussion on the thread focused on the draft
+    BIP.  Most of it focused on whether or not destruction of quantum
```
```suggestion
    BIP.  Most of it focused on whether or not destroying quantum
```
or, "whether or not the destruction of"

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    BIP119 was updated by its author to remove most (and possibly all)
+    of the controversial language; see [Newsletter #347][news347 bip119]
+    for our summary of the change and the [older version][bip119
+    prechange] for BIP119 for reference.  Some of the topics discussed
```
```suggestion
    prechange] of BIP119 for reference.  Some of the topics discussed
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    prechange] for BIP119 for reference.  Some of the topics discussed
+    included:
+
+    - *CTV+CSFS allows creation of a perpetual covenant:*
```
```suggestion
    - *CTV+CSFS allows the creation of a perpetual covenant:*
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      CTV's motivation said, "Covenants have historically been widely
+      considered to be unfit for Bitcoin because they are too complex to
+      implement and risk reducing the fungibility of coins bound by
+      them.  This BIP introduces a simple covenant called a *template*
```
```suggestion
      them.  This BIP introduces a simple covenant called a *template*,
```

harding replied:
Leaving as-is.  Quoted text; I sometimes slightly clean up punctuation in quotes but here we make a big deal about preserving the emphasis in the quote, so I want it verbatim.

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      [argument][rubin enumeration] the author of CTV (Jeremy Rubin)
+      made in 2022, where he called the type of pay-to-self covenant
+      Towns designed "recursive but fully enumerated".  Towns
+      [countered][towns enum] that the addition of CSFS undermines the
```
```suggestion
      [countered][towns enum] that adding CSFS undermines the
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      DLC service providers, including one that [claims][lava 30m] to
+      have raised "$30M in financing".
+
+    - *Vaults:* CTV simplifies implementation of [vaults][topic vaults]
```
```suggestion
    - *Vaults:* CTV simplifies the implementation of [vaults][topic vaults]
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      script described above, indicating a lack of readiness for
+      deployment.  Osuntokun [said][osuntokun enum] the tooling he uses
+      is "pretty straight forward".  Neither Towns nor Osuntokun
+      mentioned what tools they used.  Nadav Ivgi [provided][ivgi minsc]
```
"Neither Towns nor Osuntokun mentioned what tools they used." had me laughing out loud.

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    - *Vaults:* CTV simplifies implementation of [vaults][topic vaults]
+      that are possible today on Bitcoin using presigned transactions
+      and (optionally) private key deletion.  Anthony Towns [argues][towns
+      vaults] that type of vault isn't very interesting.  James O'Beirne
```
```suggestion
      vaults] that this type of vault isn't very interesting.  James O'Beirne
```

murchandamus replied:
Maybe:
```suggestion
      vaults] that this type of vault isn't very interesting.  James O'Beirne
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+    - *Accountable computing contracts:* CSFS can eliminate many steps
+      in [accountable computing contracts][topic acc] such as BitVM by
+      replacing their current need to perform Script-based lamport
+      signatures.  CTV many be able to reduce some additional signature
```
```suggestion
      signatures.  CTV may be able to reduce some additional signature
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      bidirectional bridging of tokens as part of Shielded [client-side
+      validation][topic client-side validation] (see [Newsletter
+      #322][news322 csv-bitvm]).  However, he also notes that neither
+      CTV or CSFS significantly improve the trust model of BitVM,
```
```suggestion
      CTV nor CSFS significantly improves the trust model of BitVM,
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      safer to add more features later, as each increase in expressivity
+      makes it less likely that we'll encounter a surprise.
+
+      Both Osuntokun and James O'Beirne [criticize][osuntokun enum] the
```
Just curious, why is sometimes the first name included and sometimes not? I first surmised that it was the first mention vs subsequent mentions, but James’s first name is used multiple times. Is James a special case because his brother also sometimes made Bitcoin news in the past? Or is the approach "full name once per news item, then last name"?

harding replied:
My general rule for linking to things and mentioning full names, is once per thing-that-might-be-read-independently.  In this case, I think the smallest likely unit of independent reading is the bullet "Criticism of CTV motivation"; we introduce Laolu's full name in the fifth paragraph of that bullet (in sub-bullet 1) and the quote above is the first mention of James's, so we use his full name there.

In the next major bullet I think might be read independently, "CTV+CSFS benefits", I reintroduce all of the names.  AntoineP's full name is in para 2 (sub-bullet 1), AJ's and James's full names are in para 3 (sub-bullet 2), etc...

That said, I might not have written (or, more likely, edited) a section in linear order, so I could accidentally overuse full names or topic links, so it never hurts to point out something that seems odd.

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      to be [moved] on a periodic basis."  After requests for
+      clarification, former Blockstream engineer Sanket Kanjalkar
+      [explained][kanjalkar liquid] that the benefit could be a
+      significant reduction in fees.  O'Beirne also [shared][poelstra
+      liquid] additional details from Andrew Poelstra, Blockstream's
+      Director of Research.
```
What did those details expound on?

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      making it possible was activated.  Quotes from two Core Lightning
+      developers (also co-authors of the paper introducing what we now
+      call LN-Symmetry) indicated that it was a priority for them.  A
+      quote from an LDK developer did not show it to be a priority.
```
Did it not show that it was a priority or did it show not to be a priority?

```suggestion
      quote from an LDK developer showed it not to be a priority.
```

harding replied:
Grr, I think I need to stop reporting on anything contained in a Twitter thread.  When I read @ajtowns's [link](https://x.com/TheBlueMatt/status/1859070516956389704), I only saw a subset of the conversation and AJ's conclusion seemed only weakly supported (at best).  I just checked again and it's clearly showing me an upthread post with Matt saying "I don't find eltoo all that interesting in terms of 'we need to get this done'".

I'll revise to use stronger language; thanks for double checking.

murchandamus replied:
Yeah, Matt has been of the position that Lightning needs a penalty for some time, so he is not particularly interested in LN Symmetry

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      call LN-Symmetry) indicated that it was a priority for them.  A
+      quote from an LDK developer did not show it to be a priority.
+
+    - *Ark:* Roose is CEO of a business building an [Ark][topic ark]
```
Stumbled here, perhaps needs an article
```suggestion
    - *Ark:* Roose is the CEO of a business building an [Ark][topic ark]
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      transfer ownership over the funds in the spender-cosigned
+      transaction for another presigned transaction with the receiver as
+      cosigner.  This extends the expiration date and eliminates the
+      ability for the server and previous sender to collude to steal the
```
```suggestion
      ability for the server and previous spender to collude to steal the
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+projects.  Please consider upgrading to new releases or helping to test
+release candidates._
+
+- [Bitcoin Core 29.0rc2][] is a release candidate for the next major
```
```suggestion
- [Bitcoin Core 29.0rc3][] is a release candidate for the next major
```

murchandamus commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+{% include snippets/recap-ad.md when="2025-04-08 15:30" %}
+{% include references.md %}
+{% include linkers/issues.md v=2 issues="31363,31278,3050,2963,3045,3670,9620" %}
+[bitcoin core 29.0rc2]: https://bitcoincore.org/bin/bitcoin-core-29.0/
```
```suggestion
[bitcoin core 29.0rc3]: https://bitcoincore.org/bin/bitcoin-core-29.0/
```

murchandamus reviewed (commented):
rc3 is out

harding commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,569 @@
+      partial signatures."
+
+      Roose would later start a separate thread for discussing the
+      benefits of CTV to users of Ark (see our summary below).
```
Leaving as-is.  Can't explain why, but I prefer the longer version.

harding commented:
Pushed remaining content (not including lede and releases/RCs) and made edits for all feedback to date.  I'll be doing my own review and editing later (and will add the lede/releses/topic entries then).  Thanks everyone!

azuchi commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+
+    2. If Alice receives multiple payments to the same output script,
+       she must either not spend any of them until she's ready to spend
+       all of them (definitely required for P2PKH and P2WSH; probably
```
```suggestion
       all of them (definitely required for P2PKH and P2WPKH; probably
```

It is followed by `also practically required for P2SH and P2WSH`, so is this `P2WPKH` instead of `P2WSH`?

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -464,7 +464,95 @@ Bitcoin's consensus rules._
+- **OP_CHECKCONTRACTVERIFY semantics:** Salvatore Ingala
+  [posted][ingala ccv] to Delving Bitcoin to describe the semantics of
+  the proposed [OP_CHECKCONTRACTVERIFY][topic matt] (CCV) opcode, link
+  to a [first draft BIP][ccv bip], and link to a [implementation
```
```suggestion
  to a [first draft BIP][ccv bip], and link to an [implementation
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -464,7 +464,95 @@ Bitcoin's consensus rules._
+  amount" example, which might be useful for a [vault][topic vaults]: a
+  transaction has one input (spending one output) worth 100 BTC and two
+  outputs, one for 70 BTC and the other for 30 BTC.  CCV is run twice
+  during validation of the input:
```
```suggestion
  during input validation:
```
Or maybe "while validating the input"
Or maybe "during the validation of the input"

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -464,7 +464,95 @@ Bitcoin's consensus rules._
+     would be returning that 70 BTC back to the same script it was
+     previously secured by.
+
+  2. The second time, it uses _default_ on index 1.  Although there's an
+     output being created at index 1 in this transaction, there's no
```
```suggestion
  2. The second time, it uses _default_ on index 1.  Although an
     output is created at index 1 in this transaction, there's no
```

harding replied:
Leaving as-is so that we use the same "output being created" term here as we use in the rest of the section.

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+  - A fix for [BIP34][] coinbase transaction uniqueness that should
+    fully prevent [duplicate transactions][topic duplicate transactions].
+
+  - Invalidation of future 64-byte transaction (calculated using
```
```suggestion
  - Invalidation of future 64-byte transactions (calculated using
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+    vulnerability][topic merkle tree vulnerabilities].
+
+  Technical replies were favorable for all but two parts of the
+  proposal.  The first objection, made in several replies, was to
```
```suggestion
  proposal.  The first objection, made in several replies, was to the
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+  _bridges_ between Bitcoin and other systems.  Sjors Provoost
+  [suggested][provoost bridge] someone implementing an
+  onchain-enforcable bridge provide code illustrating the difference
+  between being able to assume 64 byte transactions don't exist and
```
```suggestion
  between being able to assume 64-byte transactions don't exist and
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+
+  The second objection was about a late change to the ideas include in
+  the BIP, which was described in a [post][poinsot nsequence] on Delving
+  Bitcoin by Poinsot.  The change is requiring blocks made after
```
```suggestion
  Bitcoin by Poinsot.  The change requires blocks made after
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+  The second objection was about a late change to the ideas include in
+  the BIP, which was described in a [post][poinsot nsequence] on Delving
+  Bitcoin by Poinsot.  The change is requiring blocks made after
+  activation of consensus cleanup to set the flag that ensures makes
+  their locktime enforcable.  As previously proposed, coinbase
```
```suggestion
  activation of consensus cleanup to set the flag that ensures
  their locktime is enforceable.  As previously proposed, coinbase
```

LarryRuane commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,768 @@
+- [Eclair #3045][] makes the `payment_secret` field in the outer onion payload
+  optional for single-part [trampoline payments][topic trampoline payments].
+  Previously, every trampoline payment included a `payment_secret`, even if
+  [multipath payments][topic multipath payments] (MPP) wasn't used. Since
```
```suggestion
  [multipath payments][topic multipath payments] (MPP) weren't used. Since
```

harding commented:
Pushed my final edits (thanks @murchandamus @LarryRuane @azuchi !), added lede, and updated releases/RCs.  Quickly reviewed @Gustavojfe section (thanks!).  Did not add topic entries, that will have to wait for when I've caught up on sleep.

At over 5,700 words, equivalent to about 16 printed pages, this will be our fourth longest newsletter ever (longer than four (out of seven) of our year-in-review newsletters).  I'm glad it's done!  Thanks everyone!

bitschmidty commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,778 @@
+      to safely allow in-round acceptance of LN payments in clArk.
+
+    - *Fewer rounds, fewer signatures, and less storage:* clArk uses
+      [musig2][topic musig], and each party needs to participate in
```
```suggestion
      [MuSig2][topic musig], and each party needs to participate in
```

bitschmidty commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,778 @@
+  having to use the alternative method for preventing merkle tree
+  vulnerabilities.
+
+  The second objection was about a late change to the ideas include in
```
```suggestion
  The second objection was about a late change to the ideas included in
```

bitschmidty commented on _posts/en/newsletters/2025-04-04-newsletter.md:
```diff
@@ -0,0 +1,778 @@
+
+- [LND 0.19.0-beta.rc1][] is a release candidate for this popular LN
+  node.  One of the major improvements that could probably use testing
+  is the new RBF-based fee bumping for cooperative closes described
+  below in the notable code changes section.
```
```suggestion
  is the new RBF-based fee bumping for cooperative closes.
```

bitschmidty reviewed (commented):
Couple small updates

bitschmidty commented:
Took @vostrnad suggestions (thank you), added a few of my own for @harding sections, resolved some PR comments, and pushed a commit for some fixups and adds for @Gustavojfe notable code segment. Squashed.

@harding epic mid-year newsletter, thank you for all the time you spent on this 💪
